src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/OIDCLoginBackingBean.java

```diff
@@ -0,0 +1,111 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.omnifaces.util.Faces;
+
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.api.OpenIDConfigBean;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import fish.payara.security.openid.api.OpenIdConstant;
+import fish.payara.security.openid.api.OpenIdContext;
+import jakarta.ejb.EJB;
+import jakarta.ejb.Stateless;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.json.JsonString;
+import jakarta.json.JsonValue;
+
+/**
+ * Backing bean of the OIDC login process. Used from the login and the
+ * callback pages.
+ */
+@Stateless
+@Named
+public class OIDCLoginBackingBean implements Serializable {
+
+    private static final Logger logger = Logger.getLogger(OAuth2LoginBackingBean.class.getName());
+
+    @EJB
+    AuthenticationServiceBean authenticationSvc;
+
+    @EJB
+    SystemConfig systemConfig;
+
+    @EJB
+    UserServiceBean userService;
+
+    @Inject
+    DataverseSession session;
+
+    @Inject
+    OAuth2FirstLoginPage newAccountPage;
+
+    @Inject
+    OpenIdContext openIdContext;
+
+    @EJB
+    OpenIDConfigBean openIdConfigBean;
+
+    /**
+     * Generate the OIDC log in link.
+     */
+    public String getLogInLink() {
+        openIdConfigBean.setTarget("JSF");
+        final String email = getVerifiedEmail(openIdContext);
+        if (email != null) {
+            setUser();
+            return SystemConfig.getDataverseSiteUrlStatic();
+        }
+        return SystemConfig.getDataverseSiteUrlStatic() + "/oidc/login";
+    }
+
+    /**
+     * View action for callback.xhtml, the browser redirect target for the OAuth2
+     * provider.
+     * 
+     * @throws IOException
+     */
+    public void setUser() {
+        final String email = getVerifiedEmail(openIdContext);
+        AuthenticatedUser dvUser = authenticationSvc.getAuthenticatedUserByEmail(email);
+        if (dvUser == null) {
+            logger.log(Level.INFO, "user not found: " + email);
+            if (!systemConfig.isSignupDisabledForRemoteAuthProvider("oidc-mpconfig")) {
+                logger.log(Level.INFO, "redirect to first login: " + email);
+                Faces.redirect("/oauth2/firstLogin.xhtml");
+            }
+        } else {
+            dvUser = userService.updateLastLogin(dvUser);
+            session.setUser(dvUser);
+            Faces.redirect("/");
+        }
+    }
+
+    public static String getVerifiedEmail(final OpenIdContext oidContext) {
+        if (oidContext.getAccessToken().isExpired()) {
```

Comment:
I'm on 0703a659f2 and I'm getting

dev_dataverse> Caused by: java.lang.NullPointerException: Cannot invoke "fish.payara.security.openid.api.AccessToken.isExpired()" because the return value of "fish.payara.security.openid.api.OpenIdContext.getAccessToken()" is null
dev_dataverse> 	at edu.harvard.iq.dataverse.authorization.providers.oauth2.OIDCLoginBackingBean.getVerifiedEmail(OIDCLoginBackingBean.java:92)

... when I click the "OpenID Connect" button at http://localhost:8080/loginpage.xhtml?redirectPage=%2Fdataverse.xhtml

Reply:
I am trying to make the JSF log in to work... The changes went deeper than I hoped for. I am debugging it now and will let you know once it is working again. Too bad that Payara does not seem to let multiple systems for OIDC work at the same time, it was blocking the already implemented one because the new one was added (it looks like some security feature or something). We may end up willing to stay with the manual code, or migrating everything to the Payara built-in feature might take longer than I initially thought. To make things worse, there is also an OAuth annotation: https://docs.payara.fish/enterprise/docs/Technical%20Documentation/Public%20API/OAuth%20Support.html. I hope that the existing OAuth code still works, I do not feel motivated to migrate it too, at least not right now.

Reply:
Fixed.